### ISSUE TITLE
handle garbled fleetsync data

### ISSUE THREAD

**Issue by caver456** [ncssar/radiolog#486]
Interesting non-fatal uncaught exception - looks like the line structure was caddywampus?

![image](https://user-images.githubusercontent.com/18752102/185797479-9be86209-7bc5-4041-9cf3-824b2f8aa1cf.png)

See if this was really an issue of garbled data.  If so, handle the exception and move on.

**Comment by caver456** [comment]
Normal CID line for 100.3015:

135430:PARSING
135430:☻I11003015100301510♥☻I11003015100301510♥
135430: line:☻I11003015100301510♥☻I11003015100301510♥
135430:CID detected (not in $PKLSH): fleet=100  dev=3015  callsign=KW-100-3015

according to https://ss64.com/ascii.html, the face is Ctrl-B and the heart is Ctrl-C, so, the expected CID packet format is

Ctrl-B
I (capital letter i)
1 (number one)
<fleet#>
<device#>
<fleet#>
<device#>
Ctrl-C

then repeat that packet once.

The garbled data from the screenshot above:
^BI0100^BI010030151003015^C

What things are wrong about this data:
- there is no second packet (a packet is defined here by ^B to start, ^C to end) - it looks like the first packet got chopped off early, and didn't restart until the next ^B
- ^BI is followed by 0 instead of 1

So the question is - could we still have gotten the useful data from this data set?  Yes, definitely.

Look for any sequences of seven digits followed by ^C.  If there are multiple of these sequences, they should be identical; but if there's just one, use it anyway.

Within that sequence, the first three digits are the fleet# and the last four are the device ID.

**Comment by caver456** [comment]
From the existing parse code, apparently the 0 following ^BI is not unprecedented:
```

			elif '\x02I' in line:
				# caller ID lines look like " I110040021004002" (first character is \x02, may show as a space)
				# " I<n>" is a prefix, n is either 0 or 1 (not sure why)
```

**Comment by caver456** [comment]
fixed in recent pull request - had a typo in the fixed number - commit message said that it fixed #482

**Comment by caver456** [comment]
Reopening.  Learned recently that the CID prefix will normally be different for BOT (\x02I1) vs EOT (\x02I0).  So, right now (after the first fix of this issue), if BOT and EOT appear in the same buffered line, the entire line is recognized as garbled data and dropped, because the two CID packets are not identical.

What we really want to check is the stuff after the prefix.
\x02I110010011001001\x03\x02I010010011001001\x03 should be recognized as two 'identical' packets even though the prefix is different.
